### PR TITLE
Secure Gallery

### LINKED ISSUE #10: Secure gallery
If Selfie was started via the secure intent it shouldn't load the image and instead show a lock icon until a photo/video is taken, only then it'll be shown.

### PR DESCRIPTION
**Pull request by luca020400**
Fixes #10 
Still needs to lock drawable

### CODE REVIEW

**Review by luk1337 — CHANGES_REQUESTED** [review]
blocking until some things are fixed.

**Comment by luca020400** [comment]
@luk1337 I think I solved them all...

Lock icon is now enormous, but can't fix that now

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt [hunk 1]
```diff
@@ -727,20 +760,51 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun openGallery() {
-        dismissKeyguardAndRun {
-            sharedPreferences.getLastSavedUri()?.let { uri ->
-                listOf(MediaStore.ACTION_REVIEW, Intent.ACTION_VIEW).forEach {
+        sharedPreferences.getLastSavedUri().let { uri ->
+            // If the Uri is null attempt to lunch non secure-gallery
```
Comment (luk1337) [inline app/src/main/java/org/lineageos/selfie/MainActivity.kt]: launch

Comment (luca020400) [reply app/src/main/java/org/lineageos/selfie/MainActivity.kt]: this is what years of building AOSP does to you, mixing lunch and launch.

File: app/src/main/res/values/themes.xml [hunk 2]
```diff
@@ -40,4 +40,10 @@
         <item name="tint">?attr/colorOnPrimary</item>
         <item name="android:background">@drawable/flip_camera_button_background</item>
     </style>
+
+    <!-- Gallery button theme -->
+    <style name="SelfieGalleryButton">
```
Comment (luk1337) [inline app/src/main/res/values/themes.xml]: unused now.

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt [hunk 3]
```diff
@@ -698,14 +727,18 @@ class MainActivity : AppCompatActivity() {
...
+                runOnUiThread {
+                    galleryButton.setImageBitmap(it)
+                }
             }
-        } ?: run {
-            galleryButton.setColorFilter(getColor(R.color.dark_grey))
+        } else if (keyguardManager.isKeyguardLocked) {
+            galleryButton.setImageResource(R.drawable.ic_lock)
+        } else {
+            galleryButton.setImageResource(0)
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/selfie/MainActivity.kt]: Shouldn't really be 0, it should be a gallery icon as that's the case of no Uri in non-secure gallery

File: app/src/main/res/drawable/gallery_button_background.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (luca020400) [inline app/src/main/res/drawable/gallery_button_background.xml]: Unused?

File: app/src/main/java/org/lineageos/selfie/ViewExt.kt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package org.lineageos.selfie
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/selfie/ViewExt.kt:7]: Copyright

File: app/src/main/res/drawable/ic_lock.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (luca020400) [inline app/src/main/res/drawable/ic_lock.xml:1]: Copyright

**Review by luk1337 — APPROVED** [review]
Seems to work

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt [hunk 7]
```diff
@@ -95,7 +95,18 @@ class MainActivity : AppCompatActivity() {
     private var extensionModeIndex = 0
     private var supportedExtensionModes = listOf(extensionMode)
 
+    // Always add the secure intents first
+    // They must be checked before the non-secure ones as they won't
+    // trigger the request to dismiss the keyguard
+    // Must be sorted by priority
+    private val secureIntentsMap = mapOf(
+        Pair(MediaStore.ACTION_REVIEW_SECURE, true),
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/selfie/MainActivity.kt]: Now that I'm looking at this, we can do
ACTION to $bool
To use the kotlin extension to create pairs

**Pull request merged**